chia/farmer/farmer.py

```diff
@@ -629,6 +629,13 @@ async def generate_login_link(self, launcher_id: bytes32) -> Optional[str]:
                     self.log.error(f"Could not find authentication sk for {pool_config.p2_singleton_puzzle_hash}")
                     continue
                 authentication_token_timeout = pool_state["authentication_token_timeout"]
```

Comment:
this looks like it would throw a `KeyError` in case the message doesn't have this field

Reply:
I believe the code in `update_pool_state` makes sure that this is present with `None` default value.